### PR TITLE
Issue #162 subject has encoding while text is not encoded

### PR DESCRIPTION
**Pull request by nabilallam**
The `rfc822_encode` method retunrs a text which includes `base64`
information while the code is plain text. This leads to a wrong
presentation of the header in some mail clients (e.g. Thunderbird).

Eather the method should encode everytext or it should return `plain text` as base64 encoding is not needed.

I prefer the plain text solution, as keeps the mail header more readable.

Any ideas, suggestions critics is welcome.

### CODE REVIEW

**Comment by arnaudmorisset** [comment]
Hello!

Thanks for the contribution.

However, following this comment: https://github.com/fewlinesco/bamboo_smtp/issues/162#issuecomment-751875407 It should be better to always encode like we did before. Can you update your request to follow this approach? If you're not available, I can do it. 🙂

Thanks.

**Comment by nabilallam** [comment]
@arnaudmorisset 
I will do it.

**Comment by nabilallam** [comment]
@arnaudmorisset , please check.
thx

**Review by arnaudmorisset — APPROVED** [review]
Good for me. 🙂 

I will wait for, at least, one of my co-worker to approve before merging.

**Review by alli5723 — APPROVED** [review]

**Review by babariviere — APPROVED** [review]

**Comment by arnaudmorisset** [comment]
Thanks for the contribution. 🙇

**Pull request merged**
